### PR TITLE
Feat: Added request event navigator [#1890]

### PR DESCRIPTION
**Pull request by Abhishek9503**
## Description

Feat: Added request event navigator [#1890]

This PR added the Events Navigation in the socket.io with  the functionality mentioned in the ticket no . #1890 
 

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build

## Have you tested locally?

- [x] 👍 yes
- [ ] 🙅 no, because I am lazy

### CODE REVIEW

File: apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts [hunk 1]
```diff
@@ -339,6 +340,18 @@ class SocketIoExplorerPageViewModel {
     this.compareRequestWithServer();
   };
 
+  /**
+   *
+   * @param _headers - request headers
+   */
```
Comment (itsmdasifraza) [inline apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts:346]: update comment

File: apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts [hunk 2]
```diff
@@ -339,6 +340,18 @@ class SocketIoExplorerPageViewModel {
...
+  public upadateEvents = async (_events: EventsValues[]) => {
+    const progressiveTab = createDeepCopy(this._tab.getValue());
+    progressiveTab.property.socketio.events = _events;
+    this.tab = progressiveTab;
+    await this.tabRepository.updateTab(progressiveTab.tabId, progressiveTab);
+    this.compareRequestWithServer();
+  };
```
Comment (itsmdasifraza) [inline apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts:353]: add check for events in compare function

File: apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts (hunk 1 above)
Comment (Abhishek9503) [reply apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts:346]: Resolved

File: apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts (hunk 2 above)
Comment (Abhishek9503) [reply apps/@sparrow-desktop/src/pages/workspace-page/sub-pages/SocketIoExplorerPage/SocketIoExplorerPage.ts:353]: Resolved

**Review by itsmdasifraza — APPROVED** [review]

**Pull request merged**